crypto/encode_decode/decoder_lib.c

```diff
@@ -779,10 +779,11 @@ static int decoder_process(const OSSL_PARAM params[], void *arg)
                            (void *)new_data.ctx, LEVEL, rv);
             } OSSL_TRACE_END(DECODER);
 
-            data->flag_construct_called = 1;
             ok = (rv > 0);
-            if (ok)
+            if (ok) {
+                data->flag_construct_called = 1;
```

Comment:
If this is correct behavior then it may be worth changing the name of the 'flag_construct_called', since technically beforehand it is doing what was described i.e. the constructor was called..

Reply:
I have no objection to @mattcaswell renaming it - I don't particularly find the current names all that enlightening.
My approval stands with or without the renaming being suggested by @slontis

Reply:
That seems to be more "refactor" than bug fix which I'd rather not do in this PR since it is aimed at backport.

Reply:
I am actually not sure if this change hasn't some other unwanted consequences. Apparently the test suite passes but there might be a performance impact or some corner cases (which we do not test) might be broken.

Reply:
I doubt it will cause much problem. There are very few places where this flag is ever read. It's read in `OSSL_DECODER_from_bio` to determine whether to add an error to the stack or not (which is actually the reason for this fix, because it was omitting adding an error in cases where it should). It's also read later on in the same function to determine whether a following decoder successfully called construct (and therefore we should stop looking).

Reply:
OK, I am fine with merging this now. But perhaps @levitte could look at the consequences when he returns from his vacation.